js/base/Exchange.js

```diff
@@ -1730,7 +1730,14 @@ module.exports = class Exchange {
          * @param {string|undefined} currencyCode unified currency code, but this argument is not required by default, unless there is an exchange (like huobi) that needs an override of the method to be able to pass currencyCode argument additionally
          * @returns {[string|undefined]} unified network code
          */
-        const networkCodesByIds = this.safeValue (this.options, 'networksById', {});
+        const defaultCodes = {
+            'ERC20': 'ETH',
+            'TRC20': 'TRX',
+            'BEP20': 'BSC',
+            'BEP2': 'BNB',
+        };
+        let networkCodesByIds = this.safeValue (this.options, 'networksById', {});
```

Comment:
@samgermain we should add the defaults inside `getDefaultOptions()` so that we don't have problems when is needed to override this method
Check this out: https://github.com/ccxt/ccxt/blob/d3618536062f36b1fa658502c387730dd0c07a0e/js/base/Exchange.js#L803

Reply:
I intentionally wrote it like this so that those 4 options are always in `networkCodesByIds`, so that if `networksById` is overwritten, the person overwriting it does not need to remember to include `BEP20`, `ERC20` ... 

If they want to overwrite these network ids for whatever reason, their implementation will override the default here, because their implementation is in the second argument to this.extend

Reply:
@samgermain the thing is, some exchange might need to override the entire method (`networkIdToCode`) and, in that case, would lose all the information hardcoded.

Second thing, `ERC20` and `TRC20` are "wrong" you can't do a flat conversion here because it depends on the currency, and `networkIdToCode/networkCodeToId` already takes that into consideration.

Please check `defaultNetworkCodeReplacements` and how it's being used.